src/yoti_common/anchor.processor.js

```diff
@@ -0,0 +1,176 @@
+  this.subType = anchorObj.subType;
+  this.signedTimeStamp = anchorObj.signedTimeStamp;
+  this.originServerCerts = anchorObj.originServerCerts;
+};
+
+YotiAnchor.prototype = {
+  getValue() { return this.value; },
+  getSubType() { return this.subType; },
+  getSignedTimeStamp() { return this.signedTimeStamp; },
+  getOriginServerCerts() { return this.originServerCerts; },
+};
+
+const YotiSignedTimeStamp = function main (timeStampObj) {
+  this.version = timeStampObj.version;
+  this.timestamp = timeStampObj.timestamp;
+};
+
+YotiSignedTimeStamp.prototype = {
+  getVersion() { return this.version; },
+  getTimestamp() { return this.timestamp; }
+};
+
+module.exports.AnchorProcessor = class AnchorProcessor {
+  /**
+   * Extract matching Attribute Anchors from list.
+   *
+   * @param anchors
+   *
+   * @returns {Array}
+   */
+  static process(anchors) {
+    // This will contain a list of anchor values as objects
+    const anchorsData = {
+      sources: [],
+      verifiers: [],
+      processedValues: { // This is used to make sure the anchors values are unique
+        sources: [],
+        verifiers: [],
+      },
+    };
+
+    let anchorObj = {
+      value: '',
+      subType: '',
+      signedTimeStamp: '',
+      originServerCerts: '',
+    };
+
+    for (let i = 0; i < anchors.length; i += 1) {
+      const protoAnchor = anchors[i];
+      const certificatesList = protoAnchor.originServerCerts;
+
+      anchorObj = Object.assign(anchorObj, protoAnchor);
+      anchorObj.signedTimeStamp = this.processSignedTimeStamp(protoAnchor.getSignedTimeStamp());
+
+      for (let n = 0; n < certificatesList.length; n += 1) {
+        const certArrayBuffer = certificatesList[n];
+        const certificateObj = AnchorProcessor.convertCertToX509(certArrayBuffer);
+        const extensionsData = certificateObj.extensions;
+        const anchorTypes = AnchorProcessor.getAnchorTypes();
+
+        Object.keys(anchorTypes).forEach((key) => {
+          const oidIndex = AnchorProcessor.findOidIndex(extensionsData, { id: anchorTypes[key] });
+          if (oidIndex !== -1) {
+            const anchorExtension = extensionsData[oidIndex];
+            const anchorValue = anchorExtension.value;
+            // Convert Anchor value from ASN.1 format to an object
+            const extensionObj = forge.asn1.fromDer(anchorValue.toString('binary'));
+            if (extensionObj) {
+              const anchorParsedValue = extensionObj.value[0].value;
+              // Make sure the anchor values are unique
+              if (!anchorsData.processedValues[key].includes(anchorParsedValue)) {
+                anchorObj.value = anchorParsedValue;
+                const originServerCerts = anchorObj.originServerCerts;
+                const serverX509Certs = AnchorProcessor.convertCertsListToX509(originServerCerts);
+                anchorObj.originServerCerts = serverX509Certs;
+                anchorsData[key].push(new YotiAnchor(anchorObj));
```

Comment:
The function I suggested at line 57 should actually return the `YotiAnchor` you instantiate here.